proto/api/v1/notification_service.proto

```diff
@@ -99,19 +75,71 @@ enum NotificationLevel {
     NOTIFICATION_LEVEL_DANGER  = 4;
 }
 
+message NotificationsFilter {
+    // For filtering notifications starting from a specific timestamp, i.e. the lower boundary.
+    google.protobuf.Timestamp from     = 1;
+
+    // For filtering notifications up until a specific timestamp, i.e. the upper boundary.
+    google.protobuf.Timestamp until    = 2;
+
+    // For filtering notifications from a specific domain.
+    string domain                      = 3;
+
+    // For filtering notifications associated with a specific resource type.
+    string resource_type               = 4;
+
+    // For filtering notifications based on their type.
+    NotificationType notification_type = 5;
+
+    // For filtering notifications based on their level.
+    NotificationLevel level            = 6;
+}
+
+message CountNotificationsRequest {
+    // For filtering the notifications based on the requested fields.
+    NotificationsFilter filter = 1;
+}
+
+message CountNotificationsResponse {
+    // The total number of notifications after filtering and deduplication.
+    int64 count = 1;
+}
+
+message GetNotificationResponse {
+    Notification notification = 1;
+}
+
+message ListNotificationsRequest {
+    // For dividing the notifications response into chunks.
+    Pagination pagination      = 1;
+
+    // For filtering the notifications based on the requested fields.
+    NotificationsFilter filter = 2;
+}
+
+message ListNotificationsResponse {
+    repeated Notification notifications = 1;
+}
+
 service NotificationService {
-    // GetNotifications returns the list of notifications. The events may be filtered by
-    // dedicated fields.
-    rpc GetNotifications(GetNotificationsRequest) returns (NotificationsResponse) {
+    // CountNotifications returns the number of notifications after filtered by requested fields.
+    rpc CountNotifications(CountNotificationsRequest) returns (CountNotificationsResponse) {
         option (google.api.http) = {
-            get: "/v1/notifications"
+            get: "/v1/notificationscount"
```

Comment:
I see we primarily follow this pattern in the code, with `v1/summary/count` being the sole exception. @md2119 any recommendations?

Reply:
I think `/v1/notifications/count` should be avoided because it conflicts with `/v1/notifications/{id}`. 

In general I think all `/*count` could probably be avoided all together if we refactor the pagination such that the pagination state is returned by the paginated API itself. Then the UI no longer needs the `/*count` endpoints to calculate the pages themselves. But that's obviously a bigger can of worms.

Reply:
Agree with @stehessel that `/v1/notifications/count` with conflict with `{id}`. Typically, we introduced the count APIs within the individual resource services. I want us to consider an alternative to introduce `/v1/counts/` service. With that, the APIs can be:
`/v1/counts/deployments`,
`/v1/counts/notifications` and so on

Reply:
Thanks folks, this makes sense.